doc/extensions/zephyr/application.py

```diff
@@ -144,61 +155,105 @@ def run(self):
-                    if comment:
-                        content.append(comment.format('Linux/macOS'))
-                    prefix = '$ZEPHYR_BASE/' if zephyr_app else ''
-                    content.append('cd {}{}'.format(prefix, cd_to))
-                elif host == "win":
-                    if comment:
-                        content.append(comment.format('Windows'))
-                    prefix = '%ZEPHYR_BASE%\\' if zephyr_app else ''
-                    backslashified = cd_to.replace('/', '\\')
-                    content.append('cd {}{}'.format(prefix, backslashified))
-            content.extend(self._mkdir(mkdir, build_dir, host,
-                                       skip_config, compact))
-            if not compact:
-                content.append('')
+        if 'west' in tools:
+            w = self._generate_west(**run_config)
+            if tool_comment:
+                paragraph = nodes.paragraph()
+                paragraph += nodes.Text(tool_comment.format('west'))
+                content.append(paragraph)
+                content.append(self._lit_block(w))
+            else:
+                content.extend(w)
+
+        if 'cmake' in tools:
+            c = self._generate_cmake(**run_config)
+            if tool_comment:
+                paragraph = nodes.paragraph()
+                paragraph += nodes.Text(tool_comment.format(
+                                    'CMake and {}'.format( generator)))
+                content.append(paragraph)
+                content.append(self._lit_block(c))
+            else:
+                content.extend(c)
+
+        if not tool_comment:
+            content = [self._lit_block(content)]
 
-        if not compact:
-            content.append('# Use cmake to configure a {}-based build system:'.format(generator.capitalize()))  # noqa: E501
-        if generator == 'make':
-            content.extend(self._generate_make(**run_config))
-        elif generator == 'ninja':
-            content.extend(self._generate_ninja(**run_config))
+        return content
 
+    def _lit_block(self, content):
         content = '\n'.join(content)
 
         # Create the nodes.
         literal = nodes.literal_block(content, content)
         self.add_name(literal)
         literal['language'] = 'console'
-        return [literal]
+        return literal
+
+
+    def _generate_west(self, **kwargs):
+        content = []
+        board = kwargs['board']
+        goals = kwargs['goals']
+        cd_to = kwargs['cd_to']
+        build_dir = kwargs['build_dir']
+        kwargs['board'] = None
+        cmake_args = self._cmake_args(**kwargs)
+        cmake_args = ' --{}'.format(cmake_args) if cmake_args != '' else ''
+        # ignore zephyr_app since west needs to run within
+        # the installation. Instead rely on relative path.
+        src = ' -s {}'.format(cd_to) if cd_to else ''
+        dst = ' -d {}'.format(build_dir) if build_dir != 'build' else ''
+
+        if 'build' in goals:
```

Comment:
Ah, hello_world only has a :run: goal, explaining why there's no output for 'west'.  A quick scan shows there are 39 .rst files using this directive with only a :run: or :menuconfig: goal that would generate no west output:
```
samples/drivers/counter/alarm/README.rst:    :goals: run
samples/net/wpanusb/README.rst:   :goals: run
samples/net/wpanusb/README.rst:   :goals: run
samples/net/sockets/echo_client/README.rst:   :goals: run
samples/net/sockets/echo_client/README.rst:   :goals: run
samples/net/sockets/echo_server/README.rst:   :goals: run
samples/net/sockets/echo_server/README.rst:   :goals: run
samples/net/dhcpv4_client/README.rst:   :goals: run
samples/net/lwm2m_client/README.rst:   :goals: run
samples/net/lwm2m_client/README.rst:   :goals: run
samples/net/telnet/README.rst:   :goals: run
samples/hello_world/README.rst:   :goals: run
samples/hello_world/README.rst:   :goals: run
samples/philosophers/README.rst:   :goals: run
samples/subsys/console/getchar/README.rst:   :goals: run
samples/subsys/console/getline/README.rst:   :goals: run
samples/subsys/logging/logger/README.rst:   :goals: run
samples/mpu/mpu_test/README.rst:   :goals: run
samples/portability/cmsis_rtos_v1/timer_synchronization/README.rst:   :goals: run
samples/portability/cmsis_rtos_v1/philosophers/README.rst:   :goals: run
samples/portability/cmsis_rtos_v2/timer_synchronization/README.rst:   :goals: run
samples/portability/cmsis_rtos_v2/philosophers/README.rst:   :goals: run
samples/boards/96b_argonkey/microphone/README.rst:    :goals: run
samples/boards/96b_argonkey/sensors/README.rst:    :goals: run
samples/synchronization/README.rst:   :goals: run
samples/display/cfb_shell/README.rst:   :goals: run
doc/guides/bluetooth/index.rst:   :goals: run
doc/guides/networking/qemu_setup.rst:   :goals: run
doc/guides/networking/native_posix_setup.rst:   :goals: run
boards/x86/qemu_x86/doc/index.rst:   :goals: run
boards/arm/qemu_cortex_m3/doc/index.rst:   :goals: run
boards/arc/em_starterkit/doc/index.rst:   :goals: menuconfig
boards/arc/iotdk/doc/index.rst:   :goals: menuconfig
boards/nios2/qemu_nios2/doc/index.rst:   :goals: run
boards/riscv32/qemu_riscv32/doc/index.rst:   :goals: run
boards/xtensa/xt-sim/doc/index.rst:   :goals: menuconfig
boards/xtensa/xt-sim/doc/index.rst:   :goals: run
boards/xtensa/qemu_xtensa/doc/index.rst:   :goals: run
tests/kernel/mem_protect/protection/README.rst:   :goals: run
```

Reply:
Thanks for catching this, will fix